File: Data Structures & Algorithms/median-of-two-sorted-arrays/submission-1.js
class Solution {
    /**
     * @param {number[]} nums1
     * @param {number[]} nums2
     * @return {number}
     */
    findMedianSortedArrays(nums1, nums2) {
        const nums = [...nums1, ...nums2]
        nums.sort((a, b) => a - b)

        const i = Math.floor(nums.length / 2)
        console.log({ nums, i })
        return nums.length % 2 === 1 ? nums[i] : (nums[i - 1] + nums[i]) / 2
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        def bsearch(l: int, r: int):
            m = l + (r - l) // 2
            if l > r:
                return - 1
            elif target == nums[m]:
                return m
            elif target < nums[m]:
                return bsearch(l , m - 1)
            else:
                return bsearch(m + 1, r)
        
        return bsearch(0, len(nums) - 1)